src/main/java/domain/piece/Piece.java

```diff
@@ -0,0 +1,56 @@
+package domain.piece;
+
+import domain.board.BoardLocation;
+import domain.board.PieceExtractor;
+import domain.board.PieceFinder;
+import java.util.List;
+
+public abstract class Piece {
+
+    protected final Team team;
+
+    public Piece(Team team) {
+        this.team = team;
+    }
+
+    public void validateMovable(BoardLocation current, BoardLocation destination, PieceExtractor pieceExtractor, PieceFinder pieceFinder) {
+        validateArrival(current, destination);
+        List<BoardLocation> allPath = createAllPath(current, destination);
+        List<Piece> pathPiece = pieceExtractor.extract(allPath);
+        validateMovePath(pathPiece);
+        pieceFinder.findByLocation(destination)
+                .ifPresent(this::validateKillable);
+    }
```

Comment:
메서드 파라미터 갯수가 많아지면 적절히 줄바꿈 하는 것도 좋아보여요.

```suggestion
        public void validateMovable(
            BoardLocation current,
            BoardLocation destination,
            PieceExtractor pieceExtractor,
            PieceFinder pieceFinder
    ) {
        validateArrival(current, destination);
        List<BoardLocation> allPath = createAllPath(current, destination);
        List<Piece> pathPiece = pieceExtractor.extract(allPath);
        validateMovePath(pathPiece);
        pieceFinder.findByLocation(destination)
                .ifPresent(this::validateKillable);
    }
```